presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplitManager.java

```diff
@@ -123,11 +123,16 @@ protected void generateSegmentSplits(
                 // segments is already shuffled
                 Iterables.partition(segments, numSegmentsInThisSplit).forEach(
                         segmentsForThisSplit -> splits.add(
-                                createSegmentSplit(connectorId, pql, expectedColumnHandles, segmentsForThisSplit, host)));
+                                createSegmentSplit(connectorId, pql, expectedColumnHandles, segmentsForThisSplit, host, getGrpcPort(host))));
```

Comment:
If I am reading this line correctly, it means that we will always be using the GRPC mode if the server supports it. Is that intentional ? If so, what does the `PinotConfig.isUseStreamingForSegmentQueries` imply ?

Perhaps don't provide the GRPC port if `!isUseStreamingForSegmentQueries` ?

Reply:
So we will always set grpc port into a pinot split.

Then in `PinotPageSourceProvider`, we will decide which SegmentPageSource to use based on the config `PinotConfig.isUseStreamingForSegmentQueries`.

Reply:
Got it. 

I think we should initialize the global GPRC port to be -1 then ... just to make sure we don't by mistake talk to port 8090 even when PinotConfig.isUseStreamingForSegmentQueries = false

Reply:
make sense, will update the logic to check grpc port > 0 in PinotPageSourceProvider for picking segment page source